python/GafferUI/FileMenu.py

```diff
@@ -188,6 +189,40 @@ def openRecent( menu ) :
 
 	return result
 
+## A function suitable as the submenu callable for a File/OpenBackups menu item. It must be invoked
+# from a menu which has a ScriptWindow in its ancestry.
+def openBackup( menu ) :
+
+	scriptWindow = menu.ancestor( GafferUI.ScriptWindow )
+	currentScript = scriptWindow.scriptNode()
+	applicationRoot = currentScript.ancestor( Gaffer.ApplicationRoot )
+
+	currentWindow = GafferUI.ScriptWindow.acquire( currentScript )
+
+	backups = GafferUI.Backups.acquire( applicationRoot, createIfNecessary = False )
+
+	if backups is not None :
+		recoveryFileName = backups.recoveryFile( currentScript )
+		if not recoveryFileName :
+			return
+
+		script = Gaffer.ScriptNode()
+		script["fileName"].setValue( recoveryFileName )
+
+		dialogue = GafferUI.BackgroundTaskDialogue( "Loading" )
+		result = dialogue.waitForBackgroundTask( functools.partial( script.load, continueOnError = True, ), currentWindow)
+		if isinstance( result, IECore.Cancelled ) :
+			return
+
+		# We are loading a backup so rename the script to the old
+		# filename so the user can resave and continue as before.
+		fileName = currentScript["fileName"].getValue()
+		script["fileName"].setValue( fileName )
+		script["unsavedChanges"].setValue( True )
+		applicationRoot["scripts"].addChild( script )
```

Comment:
It seems a bit odd that this leaves you in a state where there are two scripts open, and two windows, both claiming to be for the same file. I'm not sure what the solution is here. Close the first window? Load the backup into the original ScriptNode, replacing the contents (like Revert to Saved does)? Load the backup into its own window, keeping its own filename?

Reply:
I don't really know either and tbh I haven't used backups at all so I'm not quite sure what the user would want/expect.

Reply:
Any thoughts on this @murraystevenson?